test/parallel/test-path-from-url.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('node:assert');
+const { fromURL } = require('node:path');
+const { describe, it } = require('node:test');
+
+
+describe('path.fromURL', { concurrency: true }, () => {
+  it('should passthrough path string', () => assert
+    .strictEqual(fromURL('/Users/node/dev'), '/Users/node/dev'));
+  it('should passthrough an empty string', () => assert
+    .strictEqual(fromURL(''), ''));
+  it('should passthrough a URL string with file schema', () => assert
+    .strictEqual(fromURL('file:///Users/node/dev'), 'file:///Users/node/dev'));
+  it('should parse a URL instance with file schema', () => assert
+    .strictEqual(fromURL(new URL('file:///Users/node/dev')), '/Users/node/dev'));
+  it('should parse a URL instance with file schema', () => assert
+    .strictEqual(fromURL(new URL('file:///path/to/file')), '/path/to/file'));
+  it('should remove the host from a URL instance', () => assert
+    .strictEqual(fromURL(new URL('file://localhost/etc/fstab')), '/etc/fstab'));
+  it('should parse a windows file URI', { skip: !common.isWindows }, () => assert
```

Comment:
This test should not be skipped, `/c:/foo.txt` is a valid path on POSIX. The expected value (the second argument of `strictEqual`) should depend on `common.isWindows` instead.

Reply:
Alternatively the test could use `win32.fromURL` to always get the Windows file format.